eclair-core/src/main/scala/fr/acinq/eclair/wire/LightningMessageTypes.scala

```diff
@@ -45,7 +45,9 @@ sealed trait HasChainHash extends LightningMessage { def chainHash: ByteVector32
 sealed trait UpdateMessage extends HtlcMessage // <- not in the spec
 // @formatter:on
 
-case class Init(features: ByteVector) extends SetupMessage
+case class Init(features: ByteVector, tlvs: TlvStream[InitTlv] = TlvStream.empty) extends SetupMessage {
+  val networks = tlvs.get[InitTlv.Networks].map(_.chainHashes).getOrElse(Nil)
```

Comment:
In theory this should be an `Option[List]` imo, but in that particular case it makes sense to equal `Nil` and `None`.

Reply:
Agreed, that was my reasoning too.